mir-module/src/main/resources/xsl/mods-seo-meta.xsl

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<xsl:stylesheet version="1.0"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:mods="http://www.loc.gov/mods/v3"
+                xmlns:xlink="http://www.w3.org/1999/xlink"
+                xmlns:xalan="http://xml.apache.org/xalan"
+                exclude-result-prefixes="xsl mods xlink xalan">
+
+    <xsl:include href="mods-utils.xsl"/>
+
+    <xsl:param name="CurrentLang" />
+
+    <xsl:template match="mods:mods" mode="seo-meta">
+        <xsl:apply-templates select="mods:abstract|mods:abstract/@xlink:href" mode="seo"/>
+    </xsl:template>
+
+    <xsl:template match="mods:abstract[not(@altFormat)] | mods:tableOfContents[not(@altFormat)]" mode="seo">
+        <xsl:if test="@xml:lang = $CurrentLang" >
+            <meta>
+                <xsl:attribute name="name">description</xsl:attribute>
+                <xsl:attribute name="content">
+                    <xsl:choose>
+                        <!-- SEO descriptions are limited to about 155 characters for desktop search and 120 characters
+                             for mobile search. -->
+                        <xsl:when test="string-length(.)>=140">
+                            <xsl:value-of select="concat(substring(.,1,140), '&amp;hellip;')" />
```

Comment:
Minor nitpick: The ellipsing is only relevant if the processed abstract actually more then 140 character, so a `>` comparison should be used.

Now that I've looked into this, maybe it would be better to use `mcrxml:shortenText` which appears to be made for exactly a situation like this. It makes the length comparison and picks _nicer_ a cutting by searching for the nearest whitespace.